test/java/seedu/addressbook/storage/StorageFileTest.java

```diff
@@ -0,0 +1,98 @@
+
+    @Test
+    public void initStorageFile_nullFilePath_exceptionThrown() throws InvalidStorageFilePathException {
+        thrown.expect(NullPointerException.class);
+        new StorageFile(null);
+    }
+
+    @Test
+    public void initStorageFile_invalidFilePath_exceptionThrown() throws InvalidStorageFilePathException {
+        //creating StorageFile with file not ending with .txt
+        thrown.expect(IllegalValueException.class);
+        new StorageFile(TEST_DATA_FOLDER + "/InvalidFileName");
+    }
+
+    @Test
+    public void load_invalidFormat_exceptionThrown() throws InvalidStorageFilePathException, StorageOperationException {
+        // The file contains valid xml data, but does not match the AddressBook class
+        StorageFile storage = getStorage("/InvalidData.txt");
+        thrown.expect(StorageOperationException.class);
+        storage.load();
+    }
+
+    @Test
+    public void load_validFormat() throws StorageOperationException, IllegalValueException {
+        AddressBook ab = getStorage("/ValidData.txt").load();
+        Person testPerson = new Person(new Name("John Doe"), 
+                                       new Phone("98765432", false),
+                                       new Email("johnd@gmail.com", false), 
+                                       new Address("John street, block 123, #01-01", false),
+                                       new UniqueTagList(Collections.emptySet()));
+        UniquePersonList list = new UniquePersonList(testPerson);
+
+        // ensure loaded AddressBook Object is properly constructed with sample data
+        assertTrue(ab.getAllPersons().equals(list));
+    }
+
+    @Test
+    public void save_nullAddressBook_exceptionThrown() throws InvalidStorageFilePathException, StorageOperationException {
+        //save null AddressBook object to StorageFile
+        thrown.expect(NullPointerException.class);
+        getStorage("/temp.txt").save(null);
+    }
+
+    @Test
+    public void save_validAddressBook() throws IllegalValueException, StorageOperationException, IOException {
+        StorageFile storage = getStorage("/temp.txt");
+        AddressBook ab = new AddressBook();
+        Person testPerson = new Person(new Name("John Doe"), 
+                                       new Phone("98765432", false),
+                                       new Email("johnd@gmail.com", false), 
+                                       new Address("John street, block 123, #01-01", false),
+                                       new UniqueTagList(Collections.emptySet()));
+        ab.addPerson(testPerson);
+        storage.save(ab);
+
+        // ensure xml data for sample data is properly structured and saved
+        assertTrue(compareFile("/temp.txt", "/ValidData.txt"));
+    }
+
+    private StorageFile getStorage(String filename) throws InvalidStorageFilePathException {
+        return new StorageFile(TEST_DATA_FOLDER + filename);
```

Comment:
Use `Paths.get()` rather than string concatenation here as well? Or use `File.separator`.

Then you don't need to prefix all your file paths with `/`.